src/components/Navigation/components/Item/Item.tsx

```diff
@@ -137,14 +128,38 @@ export class BaseItem extends React.Component<CombinedProps, State> {
       )
     );
 
+    let badgeMarkup: Props['badge'] = null;
+    if (isNew) {
+      badgeMarkup = (
+        <Badge status="new" size="small">
+          {intl.translate('Polaris.Badge.STATUS_LABELS.new')}
+        </Badge>
+      );
+    }
+
+    if (badge != null && !isNew) {
+      if (typeof badge === 'string') {
+        badgeMarkup = (
+          <Badge status="new" size="small">
+            {badge}
+          </Badge>
+        );
+      } else {
+        badgeMarkup = badge;
+      }
+    }
+
     const itemContentMarkup = (
       <React.Fragment>
         {iconMarkup}
         <span className={styles.Text}>
           {label}
           {indicatorMarkup}
         </span>
-        {badgeMarkup}
+
+        {badgeMarkup == null ? null : (
```

Comment:
I think I'd store this chunk in an intermediate variable, or manually wrap each case of the badge in the div when creating `badgeMarkup`